### ISSUE TITLE
`zfs list` doesn't work with 0.8.0-rc1

### ISSUE THREAD

**Issue by gregwalters** [zfsonlinux/zfs#8003]
### System information
Type | Version/Name
 --- | --- 
Distribution Name	|  CentOS
Distribution Version	| 7.5.1804
Linux Kernel	| 3.10.0-862.3.2.el7.x86_64
Architecture	| x86_64
ZFS Version	| 0.8.0-rc1
SPL Version	| 0.8.0-rc1

### Describe the problem you're observing

I'm upgrading from git (29445fe3a03d58d5aa725bd6c8964df8c8081385) to the 0.8.0-rc1 version in the kmod-testing yum repo and am unable to ```zfs list```. Maybe related to #7960?

### Describe how to reproduce the problem
Run ```zfs list```

### Include any warning/errors/backtraces from the system logs
```# zfs list
bad property list: invalid property 'name'
usage: (removed usage help message)```

```# yum whatprovides /usr/sbin/zfs
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: ftpmirror.your.org
 * epel: mirror.steadfastnet.com
 * extras: ftpmirror.your.org
 * ius: muug.ca
 * updates: repos.lax.quadranet.com
zfs-0.8.0-rc1.el7.centos.x86_64 : Commands to control the kernel modules and libraries
Repo        : @zfs-testing-kmod
Matched from:
Filename    : /usr/sbin/zfs```

```# md5sum /usr/sbin/zfs
036506eeea63f134cdb4b0ec0a91e2db  /usr/sbin/zfs```

**Comment by gregwalters** [comment]
After everything got pretty busticated I figured I might as well ```yum update``` the rest of my system and reboot. After doing the reboot I can confirm the correct module is loaded (sadly I didn't grab that before the reboot) and all of my drives are faulted with corrupted data:

```
# zpool import -d /dev/
   pool: tank
     id: 13265555963742686143
  state: UNAVAIL
 status: One or more devices contains corrupted data.
 action: The pool cannot be imported due to damaged devices or data.
   see: http://zfsonlinux.org/msg/ZFS-8000-5E
 config:

        tank        UNAVAIL  insufficient replicas
          raidz2-0  UNAVAIL  insufficient replicas
            sdk     FAULTED  corrupted data
            sdl     FAULTED  corrupted data
            sdd     FAULTED  corrupted data
            sdj     FAULTED  corrupted data
            sdg     FAULTED  corrupted data
            sdi     FAULTED  corrupted data
            sdh     FAULTED  corrupted data
            sdc     FAULTED  corrupted data
            sda     FAULTED  corrupted data
            sdb     FAULTED  corrupted data
```

I suspect I'll have to rebuild from a backup.

**Comment by behlendorf** [comment]
@gregwalters simply having a mismatched user and kernel space wouldn't have resulted in any damage to the pool itself.  Did something else happen?

**Comment by gregwalters** [comment]
No, aside from a few trials of ```zpool upgrade```, the OS update and a reboot. Right now I'm recompiling the git version I had checked out and hopefully will be able to import the pool.

**Comment by gregwalters** [comment]
After the recompile and such I'm able to access my pool again, no corruption. I'm doing another backup and will try the upgrade again in the future after ensuring that kernel and userspace, etc match. Closing this as a "mea culpa" and I need to get my stuff together next time around.

**Comment by bunder2015** [comment]
I'm having the same problem upgrading from 55972a67 to e871a8f, I need to run the zfs/zpool commands to build my new initrd using dracut so I can reboot into the new modules, but dracut has the same problem.

```
dracut: zfsexpandknowledge: pool g-pool cannot be listed: bad property list: invalid property 'name'                                                                                                                                         
usage:                                                                                                                                                                                                                                       
        list [-gHLpPv] [-o property[,...]] [-T d|u] [pool] ...                                                                                                                                                                               
            [interval [count]]                                                                                                                                                                                                               
                                                                                                                                                                                                                                             
the following properties are supported:                                                                                                                                                                                                      
                                                                                                                                                                                                                                             
        PROPERTY             EDIT   VALUES                                                                                                                                                                                                   
                                                                                                                                                                                                                                             
        feature@...           YES   disabled | enabled | active                                                                                                                                                                              
                                                                                                                                                                                                                                             
The feature@ properties must be appended with a feature name.                                                                                                                                                                                
See zpool-features(5).
```

**Comment by behlendorf** [comment]
@bunder2015 it sounds like we may need to add an override, or some compatibility logic, for the new utilities to use the old kmods.  If you can use the only zfs/zpool binaries those should work.

**Comment by bunder2015** [comment]
I tried using the 55972a6 binaries and got 

```
dracut: zfsexpandknowledge: pool g-pool cannot be listed: /sbin/zpool: Symbol `spa_feature_table' has different size in shared object, consider re-linking
```

Trying to walk through commits to see if I can find one that works.

**Comment by behlendorf** [comment]
That's a little different than the issue I was worried about here.  I was expecting e8bcb693d6a19962291e4429f2dc4ce60693bef7 to have caused this issue due to the addition of the sysfs attributes.  But 55972a6 predates that change, and I'm not quite sure what to make of the `spa_feature_table` symbol warning.

**Comment by bunder2015** [comment]
Yeah it looks like e8bcb69 is the one holding me back.

**Comment by behlendorf** [comment]
@bunder2015 can you try making this small change to 0.8-rc1 and then testing the new utilities with the old kmods.  Assuming this works as expected we can look in to making it smarter about disabling these additional checks for anything older than 0.8.

```sh
ZFS_SYSFS_IGNORE="yes" zfs list
```

```diff
diff --git a/module/zcommon/zprop_common.c b/module/zcommon/zprop_common.c
index 4d856cec1ebe..c6c61d69f9fb 100644
--- a/module/zcommon/zprop_common.c
+++ b/module/zcommon/zprop_common.c
@@ -81,6 +81,9 @@ zfs_mod_supported_prop(const char *name, zfs_type_t type)
 #if defined(_KERNEL) || defined(LIB_ZPOOL_BUILD)
        return (B_TRUE);
 #else
+       if (libzfs_envvar_is_set("ZFS_SYSFS_IGNORE"))
+               return (B_TRUE);
+
        return (zfs_mod_supported(type == ZFS_TYPE_POOL ?
            ZFS_SYSFS_POOL_PROPERTIES : ZFS_SYSFS_DATASET_PROPERTIES, name));
 #endif
```

**Comment by bunder2015** [comment]
Not sure what happened, but I'm able to build head and make a new initrd now.  I built your patch but never got to use it.  I should mention I was also upgrading from 4.14.66 to 4.19.0.  I'll close this for now.  Thanks

**Comment by bunder2015** [comment]
Afraid I gotta open this again.  Writing this down so I remember as well because I have a few more machines to upgrade.  I'll try to test your patch on another machine tomorrow.

Upgrade modules to e8bcb69, userland to bb91178, rebuild initrd and reboot, after reboot you can upgrade both to head again.

**Comment by bunder2015** [comment]
Actually I'm not sure I can test that patch since `zfs_mod_supported_prop` doesn't exist on the old kmods.  It got added in e8bcb69.

**Comment by behlendorf** [comment]
@bunder2015 you'd only need to apply that change to the new source you're building.  The issue is that the new utilities expect the kmods to create a sysfs entry which is missing in the older kmods.  This change just let's the new code optionally ignore it.  But if you can't test it that's fine, we'll put something a little smarter together.

**Comment by bunder2015** [comment]
I think I misunderstood where the patch was going.  I applied it to the userland build and it works as expected.

**Comment by bunder2015** [comment]
should be fixed in 0b8e441
